src/drivers/Cst816s.cpp

```diff
@@ -21,29 +21,19 @@ Cst816S::Cst816S(TwiMaster& twiMaster, uint8_t twiAddress) : twiMaster {twiMaste
 bool Cst816S::Init() {
   nrf_gpio_cfg_output(PinMap::Cst816sReset);
   nrf_gpio_pin_clear(PinMap::Cst816sReset);
-  vTaskDelay(5);
+  vTaskDelay(10);
   nrf_gpio_pin_set(PinMap::Cst816sReset);
   vTaskDelay(50);
 
-  // Wake the touchpanel up
-  uint8_t dummy;
-  twiMaster.Read(twiAddress, 0x15, &dummy, 1);
-  vTaskDelay(5);
-  twiMaster.Read(twiAddress, 0xa7, &dummy, 1);
-  vTaskDelay(5);
-
-  // TODO This function check that the device IDs from the controller are equal to the ones
-  // we expect. However, it seems to return false positive (probably in case of communication issue).
-  // Also, it seems that some users have pinetimes that works correctly but that report different device IDs
-  // Until we know more about this, we'll just read the IDs but not take any action in case they are not 'valid'
-  CheckDeviceIds();
+  // Read the device ids, and use the chip ID to know which variant is used.
+  ReadDeviceIds();
 
   /*
   [2] EnConLR - Continuous operation can slide around
   [1] EnConUD - Slide up and down to enable continuous operation
   [0] EnDClick - Enable Double-click action
   */
-  static constexpr uint8_t motionMask = 0b00000101;
+  static constexpr uint8_t motionMask = 0b00000111;
```

Comment:
is this something specific to the p8b? Otherwise the swipe up and down on the pinetime would be broken? Just curious

Reply:
The  register fileds `EnConLR` and `EnConUD` (should) control the "sliding" operation, i.e. event generation while a gesture is performed. LR: Left and right gestures, UD: Up and down gestures. The translation of the field description is odd at best, from what I understand the old code would only enable sliding gestures in the left and right direction. In reality, all directions ware enabled anyways before as well. Either this is a documentation mistake, or a chip behavior issue. Since the canonical way - according to the datasheet - would be to set both bits, I changed it. This might be required for some CST variants. Research is ongoing.

Register map: [CST816.pdf](https://github.com/InfiniTimeOrg/InfiniTime/files/8384517/CST816.pdf) , translation: [CST816S寄存器说明-20190508.zh-CN.en.pdf](https://github.com/InfiniTimeOrg/InfiniTime/files/8384520/CST816S.-20190508.zh-CN.en.pdf)

Reply:
The discussion can be continued in https://github.com/InfiniTimeOrg/InfiniTime/pull/1130 .